controllers/disruption_rollout_controller.go

```diff
@@ -0,0 +1,247 @@
+	}
+
+	if !targetResourceExists {
+		r.log.Warnw("target does not exist, this disruption rollout will be deleted if that continues", "error", err)
+
+		if instance.Status.TargetResourcePreviouslyMissing == nil {
+			r.log.Warnw("target is missing for the first time, updating status")
+
+			return targetResourceExists, disruptionRolloutDeleted, r.handleTargetResourceFirstMissing(ctx, instance)
+		}
+
+		if time.Since(instance.Status.TargetResourcePreviouslyMissing.Time) > TargetResourceMissingThreshold {
+			r.log.Errorw("target has been missing for over one day, deleting this schedule",
+				"timeMissing", time.Since(instance.Status.TargetResourcePreviouslyMissing.Time))
+
+			disruptionRolloutDeleted = true
+
+			return targetResourceExists, disruptionRolloutDeleted, r.handleTargetResourceMissingPastExpiration(ctx, instance)
+		}
+	} else if instance.Status.TargetResourcePreviouslyMissing != nil {
+		r.log.Infow("target was previously missing, but now present. updating the status accordingly")
+
+		return targetResourceExists, disruptionRolloutDeleted, r.handleTargetResourceNowPresent(ctx, instance)
+	}
+
+	return targetResourceExists, disruptionRolloutDeleted, nil
+}
+
+// handleTargetResourceFirstMissing handles the scenario when the target resource is missing for the first time.
+// It updates the status of the DisruptionRollout instance.
+func (r *DisruptionRolloutReconciler) handleTargetResourceFirstMissing(ctx context.Context, instance *chaosv1beta1.DisruptionRollout) error {
+	instance.Status.TargetResourcePreviouslyMissing = &metav1.Time{Time: time.Now()}
+	if err := r.Client.Status().Update(ctx, instance); err != nil {
+		return fmt.Errorf("failed to update status: %w", err)
+	}
+
+	return nil
+}
+
+// handleTargetResourceMissingPastExpiration handles the scenario when the target resource has been missing for more than the expiration period.
+// It deletes the DisruptionRollout instance.
+func (r *DisruptionRolloutReconciler) handleTargetResourceMissingPastExpiration(ctx context.Context, instance *chaosv1beta1.DisruptionRollout) error {
+	if err := r.Client.Delete(ctx, instance); err != nil {
+		return fmt.Errorf("failed to delete instance: %w", err)
+	}
+
+	return nil
+}
+
+// handleTargetResourceNowPresent handles the scenario when the target resource was previously missing but is now present.
+// It updates the status of the DisruptionRollout instance.
+func (r *DisruptionRolloutReconciler) handleTargetResourceNowPresent(ctx context.Context, instance *chaosv1beta1.DisruptionRollout) error {
+	instance.Status.TargetResourcePreviouslyMissing = nil
+	if err := r.Client.Status().Update(ctx, instance); err != nil {
+		return fmt.Errorf("failed to update status: %w", err)
+	}
+
+	return nil
+}
+
+// targetResourceUpdated checks whether the target resource has been updated or not.
+func (r *DisruptionRolloutReconciler) targetResourceUpdated(status *chaosv1beta1.DisruptionRolloutStatus) bool {
```

Comment:
This seems to be a stub? We don't check if the container hashes have changed

Reply:
This check primary purpose is to ensure that the instance.Status.LastContainerChangeTime was set, so that when we check if "target resource update has already been tested" we don't try to compare a value that has not been set.

Reply:
Correct